core/crypto/nike/nike.go

```diff
@@ -46,18 +54,21 @@ type PublicKey interface {
 	Blind(blindingFactor []byte) error
 }
 
-// Nike is an interface encapsulating a
+// Scheme is an interface encapsulating a
 // non-interactive key exchange.
-type Nike interface {
+type Scheme interface {
+
+	// Name returns the name of the NIKE scheme implementation.
+	Name() string
 
 	// PublicKeySize returns the size in bytes of the public key.
 	PublicKeySize() int
 
 	// PrivateKeySize returns the size in bytes of the private key.
 	PrivateKeySize() int
 
-	// NewKeypair returns a newly generated key pair.
-	NewKeypair() (PrivateKey, PublicKey)
+	// GenerateKeyPair creates a new key pair.
+	GenerateKeyPair() (PublicKey, PrivateKey, error)
```

Comment:
It would be nice if this took a `io.Reader` argument so it can be called/tested deterministically instead of using `rand.Reader` under the hood.
Same goes for `Blind()` which also calls `rand.Reader`

Reply:
i completely agree. however there might be some value in keeping this method as well since it conforms to the cloudflare circl's KEM Scheme interface https://github.com/cloudflare/circl/blob/main/kem/kem.go#L38

What I'm saying is that it was 100 % an intentional act to make our NIKE interfaces very similar to the KEM interfaces.

Reply:
hopefully this is acceptable to you:
```
	// GenerateKeyPair creates a new key pair.
	GenerateKeyPair() (PublicKey, PrivateKey, error)

	// GenerateKeyPairFromEntropy creates a new key pair from the given entropy source.
	GenerateKeyPairFromEntropy(rng io.Reader) (PublicKey, PrivateKey, error)

```

Reply:
OKay I've added the other constructor and I made it work for CTIDH C bindings